sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -127,6 +127,14 @@ object SQLConf {
     }
   }
 
+  val OPTIMIZER_EXCLUDED_RULES = buildConf("spark.sql.optimizer.excludedRules")
```

Comment:
If we allow this here, this will affect Spark's caching/uncaching plans and tables inconsistently. For the purpose of this PR, `StaticSQLConf.scala` would be a perfect place for this.

Reply:
Are you talking about SQL cache? I don't think optimizer has anything to do with SQL cache though, since the logical plans used to match cache entries are "analyzed" plans not "optimized" plans.

Reply:
Since an optimizer should not change query semantics(results), it should work well for the case @dongjoon-hyun described. If this is mainly used for debugging uses, I think it would be nice to use this conf on runtime.

Reply:
+1 on debugging purpose. Still, CacheManager matches the *analyzed* plan not the optimized plan.